https://learn.inside.dtu.dk/d2l/le/content/186439/viewContent/726161/View

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [17]:
Students = [120, 120]
Active = [22, 27]
Conversion = [0.183, 0.225]
df = pd.DataFrame(data=[Students, Active, Conversion], columns=["Module A", "Module B"])
df

Module A  Module B
0   120.000   120.000
1    22.000    27.000
2     0.183     0.225

Investigate a possible difference in the modules using a statistical test. Give an interpretation of
the hypothesis of no difference.

In [22]:
import statsmodels.api as sm
import math
import scipy

In [23]:
successes = df.iloc[1]
trials = df.iloc[0]
z_stat, p_value = sm.stats.proportions_ztest(successes, trials)
p_value

0.42331487293262593

In [31]:
p = 0.183
n = 120
alpha = 0.235
z = abs(scipy.stats.norm.ppf(alpha / 2))
p - z * math.sqrt((p*(1-p))/n), p + z * math.sqrt((p*(1-p))/n)

(0.14108129820059764, 0.22491870179940235)

No significant difference using propertions test. Approve the nul hypotheses of no difference

-------

If we conclude that Module B is better than Module A, what is the probability that this is a
false conclusion?

Bootstrap

This would, in this case, be wrong, as the p-value is larger than alpha

42%?, 23.5%?

-------

On the other hand, if we conclude that there is no difference between Modules A and B, what is
the probability that this is a false conclusion?

This is true, 5% probability of being wrong

------

Could the platform have used the average time spent on each learning module as the performance
metric in a comparison instead of the conversion rate? Argue why or why not.

Yes, can do p-test on if there is a difference, would probably be better to test on.
Conversion is a bit vaguely described

Could offer some problems as to what a good time is

If use a lot of time could both be too difficult or interesting by student, so could both be good and bad at the same time - conversion is easier to conclude from

--------

What are the sources that can cause violation of assumptions in your test?

Paired t-test assumptions

Subjects must be independent. Measurements for one subject do not affect measurements for any other subject.

Each of the paired measurements must be obtained from the same subject. ...

The measured differences are normally distributed.

The binomial distribution is in effect. Fixed number of trials. Trials are independent. Two categories for outcomes. 

Probabilities constant for each trial.

There are at least 5 successes and at least 5 failures.

np≥5 n p ≥ 5 and nq≥5.

------

Consider that the type of course (mathematics vs. literature) is a possible confounder for the
conversion rate. Investigate how you can take this confounder into account when comparing the
two learning modules using a statistical test.

Kruskal wallis

-------